File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        ans = 0

        for i, ch in enumerate(s):
            reversed_position = ord('z') - ord(ch) + 1
            string_position = i + 1

            ans += reversed_position * string_position

        return ans

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.cpp
class Solution {
public:
    int reverseDegree(string s) {
        int ans = 0;

        for (int i = 0; i < s.size(); i++) {
            int reversedPosition = 'z' - s[i] + 1;
            int stringPosition = i + 1;

            ans += reversedPosition * stringPosition;
        }

        return ans;
    }
};